<?xml version="1.0" encoding="utf-8"?>	

<launch>
    
    <!-- this runs on core for avoiding message looping; it muxes messages from prefixed topics "/ugvi/topic" into a single "/core/topic" -->
    
    <arg name="prefix1" default="ugv1"/>
    <arg name="prefix2" default="ugv2"/>
    <arg name="prefix3" default="ugv3"/>
      
    <node pkg="tf_remapper" type="tf_listener_and_publisher" name="tf_listener_and_publisher" output="screen">
        
        <param name = "map_frame" value = "/map"/>
        <param name = "robot_frame_id1" value = "/$(arg prefix1)/base_link"/>
        <param name = "robot_frame_id2" value = "/$(arg prefix2)/base_link"/>
        <param name = "robot_frame_id3" value = "/$(arg prefix3)/base_link"/>
        
        
        <!-- Output -->
        <remap from = "/ugv1/stamped_transform" to = "/$(arg prefix1)/stamped_transform"/> 
        <remap from = "/ugv2/stamped_transform" to = "/$(arg prefix2)/stamped_transform"/> 
        <remap from = "/ugv3/stamped_transform" to = "/$(arg prefix3)/stamped_transform"/> 
    </node>

</launch>
